### PR TITLE
[[AD]] - Pay 3504 CVE and Spring boot changes

### PR DESCRIPTION
**Pull request by AnilDhawaleGitHub**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -66,18 +66,31 @@ dependencies {
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
   compile group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-hystrix', version: '2.1.2.RELEASE'
-  compile group: 'org.springframework.hateoas', name: 'spring-hateoas', version:'0.25.1.RELEASE'
+/*  compile group: 'org.springframework.hateoas', name: 'spring-hateoas', version:'0.25.1.RELEASE'*/
```
Comment (YasharRahvar) [inline api/build.gradle:69]: Remove comments. Make sure code is clean

File: api/build.gradle [hunk 2]
```diff
@@ -66,18 +66,31 @@ dependencies {
...
   compile group: 'org.apache.commons', name: 'commons-lang3', version:'3.9'
   compile group: 'net.logstash.logback', name: 'logstash-logback-encoder', version:'5.0'
-  compile group: 'org.apache.httpcomponents', name: 'httpclient', version:'4.5.11'
-  compile group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version:'2.10.0'
+    compile group: 'org.apache.httpcomponents', name: 'httpclient', version:'4.5.11'
```
Comment (YasharRahvar) [inline api/build.gradle:72]: Code is messy. Make sure it clean and well-formatted.

File: api/build.gradle [hunk 3]
```diff
@@ -123,7 +139,11 @@ dependencies {
     compile group: 'com.github.mifmif', name: 'generex', version: '1.0.2'
 
     compile group: 'org.pitest', name: 'pitest', version: '1.3.2'
-    compile 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.3.0'
+
+    //compile 'info.solidsoft.gradle.pitest:gradle-pitest-plugin:1.3.0'
```
Comment (YasharRahvar) [inline api/build.gradle:143]: Remove comments. Make sure code is clean

File: api/src/main/java/uk/gov/hmcts/payment/api/dto/mapper/CreditAccountDtoMapper.java [hunk 4]
```diff
@@ -208,12 +208,12 @@ public FeeDto toFeeDto(PaymentFee fee) {
     @SneakyThrows(NoSuchMethodException.class)
     private PaymentDto.LinkDto cancellationLink(String userId, Integer paymentId) {
         Method method = CreditAccountPaymentController.class.getMethod("cancel", String.class, Integer.class);
-        return new PaymentDto.LinkDto(ControllerLinkBuilder.linkTo(method, userId, paymentId).toString(), "POST");
+        return new PaymentDto.LinkDto(WebMvcLinkBuilder.linkTo(method, userId, paymentId).toString(), "POST");
```
Comment (YasharRahvar) [inline api/src/main/java/uk/gov/hmcts/payment/api/dto/mapper/CreditAccountDtoMapper.java:211]: check why codecov is complaining that this line is not covered

File: api/src/main/java/uk/gov/hmcts/payment/api/v1/controllers/PaymentDtoFactory.java [hunk 5]
```diff
@@ -32,7 +32,7 @@ public PaymentOldDto toDto(PaymentOld paymentOld) {
     @SneakyThrows(NoSuchMethodException.class)
     private PaymentOldDto.LinkDto cancellationLink(String userId, Integer paymentId) {
         Method method = PaymentOldController.class.getMethod("cancel", String.class, Integer.class);
-        return new PaymentOldDto.LinkDto(ControllerLinkBuilder.linkTo(method, userId, paymentId).toString(), "POST");
+        return new PaymentOldDto.LinkDto(WebMvcLinkBuilder.linkTo(method, userId, paymentId).toString(), "POST");
```
Comment (YasharRahvar) [inline api/src/main/java/uk/gov/hmcts/payment/api/v1/controllers/PaymentDtoFactory.java:35]: Check why codecov is complaining

File: api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CaseControllerTest.java [hunk 6]
```diff
@@ -414,11 +414,11 @@ public void getAllPaymentGroupsHavingMultipleFeesRemissionsAndPaymentsWithCcdCas
         PaymentFeeLink paymentFeeLink = paymentDbBackdoor.findByReference(createPaymentResponseDto.getPaymentGroupReference());
         PaymentFee fee = paymentFeeDbBackdoor.findByPaymentLinkId(paymentFeeLink.getId());
 
-        // create a partial remission
+      /*  // create a partial remission
```
Comment (YasharRahvar) [inline api/src/test/java/uk/gov/hmcts/payment/api/componenttests/CaseControllerTest.java:417]: Why we have committed out these tests?

File: api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java [hunk 7]
```diff
@@ -58,7 +68,10 @@ public CacheManager cacheManager() {
                     .build()),
             new CaffeineCache("sites", Caffeine.newBuilder()
                 .expireAfterWrite(48, TimeUnit.HOURS)
-                .build())
+                .build()),
```
Comment (aravindan89) [inline api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java:71]: @amitchidrewar1301 @AnilDhawaleGitHub -  I think this was added as part of OIDC changes. I don't think so we need this change for spring boot upgrade.

File: api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java [hunk 8]
```diff
@@ -58,7 +68,10 @@ public CacheManager cacheManager() {
...
+        new CaffeineCache("userInfoCache", Caffeine.newBuilder()
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java:72]: @AnilDhawaleGitHub Did we have a requirement on Caching?

File: api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java (hunk 7 above)
Comment (AnilDhawaleGitHub) [reply api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java:71]: removed

File: api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java (hunk 8 above)
Comment (AnilDhawaleGitHub) [reply api/src/main/java/uk/gov/hmcts/PaymentApiApplication.java:72]: no we have removed it

**Comment by aravindan89** [comment]
Closing this as these changes are already pushed by @AnilDhawaleGitHub to prod

**Pull request closed without merging**